src/Microsoft.CodeQuality.Analyzers/Core/Security/MarkVerbHandlersWithValidateAntiforgeryTokenAnalyzer.cs

```diff
@@ -0,0 +1,360 @@
+using System;
+
+namespace Microsoft.CodeQuality.Analyzers.Security
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public class MarkVerbHandlersWithValidateAntiforgeryTokenAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string RuleId = "CA3147";
+
+        private static readonly LocalizableString Title = new LocalizableResourceString(
+            nameof(MicrosoftSecurityAnalyzersResources.MarkVerbHandlersWithValidateAntiforgeryTokenTitle),
+            MicrosoftSecurityAnalyzersResources.ResourceManager,
+            typeof(MicrosoftSecurityAnalyzersResources));
+
+        private static readonly LocalizableString NoVerbsMessage = new LocalizableResourceString(
+            nameof(MicrosoftSecurityAnalyzersResources.MarkVerbHandlersWithValidateAntiforgeryTokenNoVerbsMessage),
+            MicrosoftSecurityAnalyzersResources.ResourceManager,
+            typeof(MicrosoftSecurityAnalyzersResources));
+
+        private static readonly LocalizableString NoVerbsNoTokenMessage = new LocalizableResourceString(
+            nameof(MicrosoftSecurityAnalyzersResources.MarkVerbHandlersWithValidateAntiforgeryTokenNoVerbsNoTokenMessage),
+            MicrosoftSecurityAnalyzersResources.ResourceManager,
+            typeof(MicrosoftSecurityAnalyzersResources));
+
+        private static readonly LocalizableString GetAndTokenMessage = new LocalizableResourceString(
+            nameof(MicrosoftSecurityAnalyzersResources.MarkVerbHandlersWithValidateAntiforgeryTokenGetAndTokenMessage),
+            MicrosoftSecurityAnalyzersResources.ResourceManager,
+            typeof(MicrosoftSecurityAnalyzersResources));
+
+        private static readonly LocalizableString GetAndOtherAndTokenMessage = new LocalizableResourceString(
+            nameof(MicrosoftSecurityAnalyzersResources.MarkVerbHandlersWithValidateAntiforgeryTokenGetAndOtherAndTokenMessage),
+            MicrosoftSecurityAnalyzersResources.ResourceManager,
+            typeof(MicrosoftSecurityAnalyzersResources));
+
+        private static readonly LocalizableString VerbsAndNoTokenMessage = new LocalizableResourceString(
+            nameof(MicrosoftSecurityAnalyzersResources.MarkVerbHandlersWithValidateAntiforgeryTokenVerbsAndNoTokenMessage),
+            MicrosoftSecurityAnalyzersResources.ResourceManager,
+            typeof(MicrosoftSecurityAnalyzersResources));
+
+        private static readonly DiagnosticDescriptor NoVerbsRule = new DiagnosticDescriptor(
+            RuleId,
+            Title, 
+            NoVerbsMessage, 
+            DiagnosticCategory.Security,
+            DiagnosticSeverity.Warning, 
+            isEnabledByDefault: false);
+
+        private static readonly DiagnosticDescriptor NoVerbsNoTokenRule = new DiagnosticDescriptor(
+            RuleId,
+            Title,
+            NoVerbsNoTokenMessage,
+            DiagnosticCategory.Security,
+            DiagnosticSeverity.Warning,
+            isEnabledByDefault: false);
```

Comment:
I see that you are turning off all these analyzer rules off by default. Is this intentional? Are these rules noisy?

Reply:
I would recommend turning it on by default if you think the rule is useful and not noisy/controversial. That is our general approach for ported rules.